client/components/data/query-active-promotions/index.jsx

```diff
@@ -1,45 +1,25 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { isRequestingActivePromotions } from 'state/active-promotions/selectors';
 import { requestActivePromotions } from 'state/active-promotions/actions';
 
-class QueryActivePromotions extends Component {
-	UNSAFE_componentWillMount() {
-		if ( ! this.props.requestingActivePromotions ) {
-			this.props.requestActivePromotions();
+export default function QueryActivePromotions() {
+	const requestingActivePromotions = useRef( useSelector( isRequestingActivePromotions ) );
```

Comment:
Is this a good idiom, to ensure that `requestingActivePromotions` doesn't change, and thus the effect below only runs on mount?

Reply:
Answered elsewhere; the preferred approach is to make use of the dependency array instead.